src/Clapi/TextSerialisation.hs

```diff
@@ -0,0 +1,33 @@
+module Clapi.TextSerialisation where
+import Data.Monoid
+import qualified Data.Map as Map
+import Blaze.ByteString.Builder (Builder)
+import Blaze.ByteString.Builder.Char.Utf8 (fromString, fromChar)
+import qualified Clapi.Serialisation as Wire
+import Clapi.Types (Time(..), ClapiValue(..))
+
+typeTag :: ClapiValue -> Char
+typeTag (ClBool _) = 'B'
+typeTag v = Wire.typeTag v
+
+fss :: Show a => a -> Builder
+fss a = fromString $ show a
+
+cvBuilder :: ClapiValue -> Builder
+cvBuilder (ClBool True) = fromChar 'T'
+cvBuilder (ClBool False) = fromChar 'F'
+cvBuilder (ClInt32 i) = fss i
+cvBuilder (ClString s) = fss s
+
+tpBuilder :: (Time, [ClapiValue]) -> Builder
+tpBuilder ((Time s f), vs) = tb <> mconcat (map (\v -> (fromChar ' ') <> (cvBuilder v)) vs)
+  where
+    tb = (fss s) <> (fromChar ':') <> (fss f)
```

Comment:
If we were to be unit pedants, isn't the separator here just `'.'`? We're just doing decimal fractions of a second, right?

Reply:
No, we're doing 1/MAX_INT32 s of a second.

Reply:
Ah, yes, duh.